# <font color='blue'>Video Game Sales Prediction 1980 - 2020</font>

## Table of Contents
1. [Reseach Problem](#a_ID)

2. [Data Preparation](#b_ID)

## <a id="a_ID">1. Research Problem</a>

Which machine learning model best predits North American video game sales?

kaggle Dataset: Video Game Sales

## <a id="b_ID">2. Data Preparation</a>

In [3]:
# Import numpy and pandas libraries for data analysis
import numpy as np
import pandas as pd

In [4]:
# Import matplotlib and seaborn libraries for data visualizations
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

In [5]:
# Read the video game sales dataset
vgsales = pd.read_csv('vgsales.csv')

In [6]:
# Find the dimension of the dataframe
print("Dataset Shape: ", vgsales.shape)

Dataset Shape:  (16598, 11)


There are 16,598 records (rows) and 11 features (columns) in this dataset.

In [7]:
# Find index and data type of the features
vgsales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


There are 11 features and the target attribute for our analysis is North American sales (NA_Sales).

In [8]:
# Summary statistics for numeric features
vgsales.describe()

Rank          Year      NA_Sales      EU_Sales      JP_Sales  \
count  16598.000000  16327.000000  16598.000000  16598.000000  16598.000000   
mean    8300.605254   2006.406443      0.264667      0.146652      0.077782   
std     4791.853933      5.828981      0.816683      0.505351      0.309291   
min        1.000000   1980.000000      0.000000      0.000000      0.000000   
25%     4151.250000   2003.000000      0.000000      0.000000      0.000000   
50%     8300.500000   2007.000000      0.080000      0.020000      0.000000   
75%    12449.750000   2010.000000      0.240000      0.110000      0.040000   
max    16600.000000   2020.000000     41.490000     29.020000     10.220000   

        Other_Sales  Global_Sales  
count  16598.000000  16598.000000  
mean       0.048063      0.537441  
std        0.188588      1.555028  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.040000      0.470000  
max       10.570000     82.740000

In [9]:
# Convert Year to appropriate categorical attribute
vgsales['Year'] = vgsales['Year'].astype(np.object)

# Confirm data types are appropriate
vgsales.dtypes

Rank              int64
Name             object
Platform         object
Year             object
Genre            object
Publisher        object
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object

In [10]:
# Look at first 5 records
vgsales.head()

Rank                      Name Platform  Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [11]:
# Look at last 5 records
vgsales.tail()

Rank                                              Name Platform  Year  \
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA  2002   
16594  16597                     Men in Black II: Alien Escape       GC  2003   
16595  16598  SCORE International Baja 1000: The Official Game      PS2  2008   
16596  16599                                        Know How 2       DS  2010   
16597  16600                                  Spirits & Spells      GBA  2003   

          Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
16593  Platform       Kemco      0.01      0.00       0.0          0.0   
16594   Shooter  Infogrames      0.01      0.00       0.0          0.0   
16595    Racing  Activision      0.00      0.00       0.0          0.0   
16596    Puzzle    7G//AMES      0.00      0.01       0.0          0.0   
16597  Platform     Wanadoo      0.01      0.00       0.0          0.0   

       Global_Sales  
16593          0.01  
16594          0.01  
16595          0.01  
16596          0.01  
16597          0.01

In [12]:
# Look at random 5 records
vgsales.sample(5, random_state = 15)

Rank                                    Name Platform  Year  \
162      163                     Super Mario Advance      GBA  2001   
8928    8930                                 F1 2012       PC  2012   
12435  12437                      Puzzler Collection      Wii  2008   
8098    8100  Sesame Street: Elmo's Letter Adventure      N64  1999   
568      569                                   Fable       XB  2004   

              Genre               Publisher  NA_Sales  EU_Sales  JP_Sales  \
162        Platform                Nintendo      3.14      1.24      0.91   
8928         Racing             Codemasters      0.01      0.11      0.00   
12435        Puzzle                 Ubisoft      0.05      0.01      0.00   
8098           Misc                NewKidCo      0.14      0.04      0.00   
568    Role-Playing  Microsoft Game Studios      1.99      0.58      0.00   

       Other_Sales  Global_Sales  
162           0.20          5.49  
8928          0.03          0.15  
12435         0.00          0.06  
8098          0.00          0.18  
568           0.09          2.66

In [13]:
# Count the number of missing values in the dataset
vgsales.isnull().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

There are 271 missing values in Year and 58 missing values in Publisher.

In [14]:
# Check the missing values in Year
vgsales[vgsales['Year'].isnull()]

Rank                         Name Platform Year         Genre  \
179      180              Madden NFL 2004      PS2  NaN        Sports   
377      378             FIFA Soccer 2004      PS2  NaN        Sports   
431      432   LEGO Batman: The Videogame      Wii  NaN        Action   
470      471   wwe Smackdown vs. Raw 2006      PS2  NaN      Fighting   
607      608               Space Invaders     2600  NaN       Shooter   
...      ...                          ...      ...  ...           ...   
16307  16310                Freaky Flyers       GC  NaN        Racing   
16327  16330                    Inversion       PC  NaN       Shooter   
16366  16369  Hakuouki: Shinsengumi Kitan      PS3  NaN     Adventure   
16427  16430                 Virtua Quest       GC  NaN  Role-Playing   
16493  16496                   The Smurfs      3DS  NaN        Action   

                                    Publisher  NA_Sales  EU_Sales  JP_Sales  \
179                           Electronic Arts      4.26      0.26      0.01   
377                           Electronic Arts      0.59      2.36      0.04   
431    Warner Bros. Interactive Entertainment      1.86      1.02      0.00   
470                                       NaN      1.57      1.02      0.00   
607                                     Atari      2.36      0.14      0.00   
...                                       ...       ...       ...       ...   
16307                                 Unknown      0.01      0.00      0.00   
16327                      Namco Bandai Games      0.01      0.00      0.00   
16366                                 Unknown      0.01      0.00      0.00   
16427                                 Unknown      0.01      0.00      0.00   
16493                                 Unknown      0.00      0.01      0.00   

       Other_Sales  Global_Sales  
179           0.71          5.23  
377           0.51          3.49  
431           0.29          3.17  
470           0.41          3.00  
607           0.03          2.53  
...            ...           ...  
16307         0.00          0.01  
16327         0.00          0.01  
16366         0.00          0.01  
16427         0.00          0.01  
16493         0.00          0.01  

[271 rows x 11 columns]

In [15]:
# Check the missing values in Publisher
vgsales[vgsales['Publisher'].isnull()]

Rank                                               Name Platform  \
470      471                         wwe Smackdown vs. Raw 2006      PS2   
1303    1305                                     Triple Play 99       PS   
1662    1664       Shrek / Shrek 2 2-in-1 Gameboy Advance Video      GBA   
2222    2224                                 Bentley's Hackpack      GBA   
3159    3161  Nicktoons Collection: Game Boy Advance Video V...      GBA   
3166    3168  SpongeBob SquarePants: Game Boy Advance Video ...      GBA   
3766    3768  SpongeBob SquarePants: Game Boy Advance Video ...      GBA   
4145    4147                                 Sonic the Hedgehog      PS3   
4526    4528  The Fairly Odd Parents: Game Boy Advance Video...      GBA   
4635    4637  The Fairly Odd Parents: Game Boy Advance Video...      GBA   
5302    5304      Dragon Ball Z: Budokai Tenkaichi 2 (JP sales)      Wii   
5647    5649  Cartoon Network Collection: Game Boy Advance V...      GBA   
6272    6274  The Legend of Zelda: The Minish Cap(weekly JP ...      GBA   
6437    6439           Sonic X: Game Boy Advance Video Volume 1      GBA   
6562    6564  Dora the Explorer: Game Boy Advance Video Volu...      GBA   
6648    6650  Cartoon Network Collection: Game Boy Advance V...      GBA   
6849    6851     All Grown Up!: Game Boy Advance Video Volume 1      GBA   
7208    7210  Nicktoons Collection: Game Boy Advance Video V...      GBA   
7351    7353                     Yu Yu Hakusho: Dark Tournament      PS2   
7470    7472  SpongeBob SquarePants: Game Boy Advance Video ...      GBA   
7953    7955                   Thomas the Tank Engine & Friends      GBA   
8330    8332    Dragon Ball GT: Game Boy Advance Video Volume 1      GBA   
8341    8343  Codename: Kids Next Door: Game Boy Advance Vid...      GBA   
8368    8370  Teenage Mutant Ninja Turtles: Game Boy Advance...      GBA   
8503    8505                                       Stronghold 3       PC   
8770    8772  Cartoon Network Collection: Game Boy Advance V...      GBA   
8848    8850  Pokémon: Johto Photo Finish: Game Boy Advance ...      GBA   
8896    8898  Strawberry Shortcake: Game Boy Advance Video V...      GBA   
9517    9519                             Farming Simulator 2011       PC   
9749    9751  Super Robot Wars OG Saga: Masou Kishin II - Re...      PSP   
10382  10384                   Disney Channel Collection Vol. 1      GBA   
10494  10496             Atsumare! Power Pro Kun no DS Koushien       DS   
11076  11078                       Action Man-Operation Extreme       PS   
11526  11528  Cartoon Network Collection: Game Boy Advance V...      GBA   
12487  12489                               Chou Soujuu Mecha MG       DS   
12517  12519       Prinny: Can I Really Be The Hero? (US sales)      PSP   
13278  13280                     Monster Hunter Frontier Online      PS3   
13672  13674                          B.L.U.E.: Legend of Water       PS   
13962  13964                                     World of Tanks     X360   
14087  14089                                       Housekeeping       DS   
14296  14299                                Bikkuriman Daijiten       DS   
14311  14314                                      Silverlicious       DS   
14698  14701                                 UK Truck Simulator       PC   
14942  14945  Umineko no Naku Koro ni San: Shinjitsu to Gens...      PS3   
15056  15059                                            Xia-Xia       DS   
15261  15264                                       Mario Tennis      3DS   
15325  15328  Nicktoons Collection: Game Boy Advance Video V...      GBA   
15353  15356                   Demolition Company: Gold Edition       PC   
15788  15791  Moshi, Kono Sekai ni Kami-sama ga Iru to suru ...      PSV   
15915  15918                                       Dream Dancer       DS   
16191  16194                    Homeworld Remastered Collection       PC   
16198  16201             AKB1/48: Idol to Guam de Koishi

In [16]:
# Count the number of duplicated records in the dataset
vgsales.duplicated().sum(axis=0) # 0 for rows and 1 for columns

0

There are no duplicate records.

In [17]:
# Find the unique values in the "Platform" column of the dataframe
print(vgsales['Platform'].unique())
print(vgsales['Platform'].nunique())

['Wii' 'NES' 'GB' 'DS' 'X360' 'PS3' 'PS2' 'SNES' 'GBA' '3DS' 'PS4' 'N64'
 'PS' 'XB' 'PC' '2600' 'PSP' 'XOne' 'GC' 'WiiU' 'GEN' 'DC' 'PSV' 'SAT'
 'SCD' 'WS' 'NG' 'TG16' '3DO' 'GG' 'PCFX']
31


There are 31 unique values in the "Platform" column.

In [18]:
# Find the unique values in the "Year" column of the dataframe
print(vgsales['Year'].unique())
print(vgsales['Year'].nunique())

[2006.0 1985.0 2008.0 2009.0 1996.0 1989.0 1984.0 2005.0 1999.0 2007.0
 2010.0 2013.0 2004.0 1990.0 1988.0 2002.0 2001.0 2011.0 1998.0 2015.0
 2012.0 2014.0 1992.0 1997.0 1993.0 1994.0 1982.0 2003.0 1986.0 2000.0 nan
 1995.0 2016.0 1991.0 1981.0 1987.0 1980.0 1983.0 2020.0 2017.0]
39


There are 39 unique values in the "Year" column (Years from 1980 - 2020).

In [19]:
# Find the unique values in the "Genre" column of the dataframe
print(vgsales['Genre'].unique())
print(vgsales['Genre'].nunique())

['Sports' 'Platform' 'Racing' 'Role-Playing' 'Puzzle' 'Misc' 'Shooter'
 'Simulation' 'Action' 'Fighting' 'Adventure' 'Strategy']
12


There are 12 unique values in the "Genre" column.

In [20]:
# Find the unique values in the "Publisher" column of the dataframe
print(vgsales['Publisher'].unique())
print(vgsales['Publisher'].nunique())

['Nintendo' 'Microsoft Game Studios' 'Take-Two Interactive'
 'Sony Computer Entertainment' 'Activision' 'Ubisoft' 'Bethesda Softworks'
 'Electronic Arts' 'Sega' 'SquareSoft' 'Atari' '505 Games' 'Capcom'
 'GT Interactive' 'Konami Digital Entertainment'
 'Sony Computer Entertainment Europe' 'Square Enix' 'LucasArts'
 'Virgin Interactive' 'Warner Bros. Interactive Entertainment'
 'Universal Interactive' 'Eidos Interactive' 'RedOctane' 'Vivendi Games'
 'Enix Corporation' 'Namco Bandai Games' 'Palcom' 'Hasbro Interactive'
 'THQ' 'Fox Interactive' 'Acclaim Entertainment' 'MTV Games'
 'Disney Interactive Studios' nan 'Majesco Entertainment' 'Codemasters'
 'Red Orb' 'Level 5' 'Arena Entertainment' 'Midway Games' 'JVC'
 'Deep Silver' '989 Studios' 'NCSoft' 'UEP Systems' 'Parker Bros.' 'Maxis'
 'Imagic' 'Tecmo Koei' 'Valve Software' 'ASCII Entertainment' 'Mindscape'
 'Infogrames' 'Unknown' 'Square' 'Valve' 'Activision Value' 'Banpresto'
 'D3Publisher' 'Oxygen Interactive' 'Red Storm Entertainmen

There are 578 unique values in the "Publisher" column.

In [21]:
# Find the number of unique values in the "Name" and "Rank" columns of the dataframe
print(vgsales['Name'].nunique())
print(vgsales['Rank'].nunique())

11493
16598


There are 11493 unique values in the "Name" column and there are 16598 unique values in the "Rank" column (same as 16598 total records).